api/tests/routes/backoffice/accounts_test.py

```diff
@@ -2452,3 +2453,77 @@ def test_anonymize_public_account_when_user_does_not_meet_criteria(
 
         response = self.post_to_endpoint(authenticated_client, user_id=user.id)
         assert response.status_code == 400
+
+
+class ExtractPublicAccountTest(PostEndpointHelper):
+    endpoint = "backoffice_web.public_accounts.create_extract_user_gdpr_data"
+    endpoint_kwargs = {"user_id": 1}
+    needed_permission = perm_models.Permissions.EXTRACT_PUBLIC_ACCOUNT
+
+    @mock.patch("pcapi.routes.backoffice.accounts.blueprint.extract_beneficiary_data.delay")
+    def test_extract_public_account(self, _mock_extract, authenticated_client, legit_user):
+
+        user = users_factories.BeneficiaryFactory()
+
+        expected_queries = 4  # session + user + user data + gdpr insert
```

Comment:
Détail : mieux pour bien distinguer les `user` : 
```suggestion
        expected_queries = 4  # session + authenticated user + targeted user with joined data + gdpr insert
```